src/domain/agreement/agreement.entity.ts

```diff
@@ -35,20 +37,15 @@ export class Agreement extends BaseEntity implements IAgreement {
   )
   project?: Project;
 
-  @Field(() => [Tag], {
-    nullable: true,
-    description: 'The set of tags for this Agreement e.g. Team, Nature etc.',
-  })
-  @ManyToMany(
-    () => Tag,
-    tag => tag.agreements,
-    { eager: true, cascade: true }
-  )
-  @JoinTable({ name: 'agreement_tag' })
-  tags?: Tag[];
+  @Field(() => Tagset, { nullable: true, description: 'The set of tags for the agreement' })
+  @OneToOne(() => Tagset, { eager: true, cascade: true })
+  @JoinColumn()
+  tagset: Tagset;
 
   constructor(name: string) {
     super();
     this.name = name;
+    this.tagset = new Tagset(RestrictedTagsetNames.Default);
+    //this.tagsetService.initialiseMembers(this.tagset);
```

Comment:
Will this be used?

Reply:
will handle the whole initialisation path as a separate issue. Good to leave in for now as a reminder.

Reply:
I saw and recalled our conversation earlier - hence I haven't commented on the rest of the files 👍